flow/record/context.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import annotations
+
+import sys
+from contextlib import contextmanager
+from contextvars import ContextVar
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from collections.abc import Generator
+
+    from flow.record import Record
+    from flow.record.selector import Selector
+
+APP_CONTEXT: ContextVar[AppContext] = ContextVar("APP_CONTEXT")
+
+
+def get_app_context() -> AppContext:
+    """Retrieve the application context, creating it if it does not exist.
+
+    Returns:
+        AppContext: The application context.
```

Comment:
```suggestion
        The application context.
```
Return type is already provided in the type hint.